packages/labs/test/selectTests.tsx

```diff
@@ -98,39 +98,127 @@ describe("<Select>", () => {
+        assert.equal(wrapper.find("input").prop("value"), query);
+        assert.equal(wrapper.state("query"), query);
+    });
+
+    it("onQueryChange is called when the input value changes", () => {
+        const onQueryChange = sinon.spy();
+        const wrapper = select({ onQueryChange });
+        const query = "nailed it";
+        const input = wrapper.find("input");
+        (input.getDOMNode() as HTMLInputElement).value = query;
+        input.simulate("change");
+        assert.isTrue(onQueryChange.calledOnce);
+        assert.isTrue(onQueryChange.calledWithExactly(query));
+    });
+
+    it("if onQueryChange and inputProps.onChange are passed, both will be called", () => {
+        const onQueryChange = sinon.spy();
+        const onInputChange = sinon.spy();
+        const wrapper = select({ onQueryChange, inputProps: { onChange: onInputChange } });
+        wrapper.find("input").simulate("change");
+        assert.isTrue(onQueryChange.calledOnce);
+        assert.isTrue(onInputChange.calledOnce);
+    });
+
+    it("onQueryChange is called when resetOnSelect=true", () => {
+        const onQueryChange = sinon.spy();
+        const wrapper = select({ onQueryChange, resetOnSelect: true, query: "1972" });
+        wrapper
+            .find("a")
+            .at(0)
+            .simulate("click");
+        assert.isTrue(onQueryChange.calledOnce);
+        assert.isTrue(onQueryChange.calledWithExactly(""));
+    });
+
+    it("onQueryChange is called when resetOnClose=true", () => {
+        const onQueryChange = sinon.spy();
+        const wrapper = select({ onQueryChange, resetOnClose: true, popoverProps: {}, query: "1972" });
+        wrapper.find("table").simulate("click");
+        assert.isTrue(wrapper.find(Popover).prop("isOpen"));
+        wrapper.find("table").simulate("click");
+        assert.isFalse(wrapper.find(Popover).prop("isOpen"));
+        assert.isTrue(onQueryChange.calledOnce);
+        assert.isTrue(onQueryChange.calledWithExactly(""));
+    });
+
+    it("onQueryChange is called when clear button pressed", () => {
+        const onQueryChange = sinon.spy();
+        const wrapper = select({ onQueryChange, query: "1972" });
+        wrapper
+            .find(InputGroup)
+            .find(Button)
+            .simulate("click");
+        assert.isTrue(onQueryChange.calledOnce);
+        assert.isTrue(onQueryChange.calledWithExactly(""));
+    });
+
     it("input can be controlled with inputProps", () => {
         const value = "nailed it";
         const onChange = sinon.spy();
 
-        const input = select({ inputProps: { value, onChange } }).find("input");
+        const wrapper = select({ inputProps: { value, onChange } });
+        assert.equal(wrapper.state("query"), value);
+
+        const input = wrapper.find("input");
         assert.equal(input.prop("value"), value);
+        assert.equal(wrapper.state("query"), value);
```

Comment:
not convinced that we need to validate that state matches input prop _everywhere_. just doing it in the relevant tests should be enough.

Reply:
good point - should be testing one thing